### PR TITLE
[Compute] `az vm/vmss create`: Support creating VM/VMSS from community gallery image and add community gallery legal agreement acceptance

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Feature request: #21815

**Description**

- Support creating VM/VMSS from community gallery image *(previously, it was only implemented in the private package from edge build, and there was no merge into the official CLI)*
- Add community gallery legal agreement acceptance
- Add the verification of whether `--os-type` is correct when creating VM from community gallery image or shared gallery image

**Testing Guide**

**History Notes**

[Compute] `az vm/vmss create`: Support creating VM/VMSS from community gallery image
[Compute] `az vm/vmss create`: Add community gallery legal agreement acceptance
[Compute] `az vm/vmss create`: Add the verification of whether `--os-type` is correct when creating VM from community gallery image or shared gallery image

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

**Comment by zhoxing-ms** [comment]
> Hey Xing, can we update the description of 'eula' in the same pr too? it is https://github.com/Azure/azure-cli/pull/3 from https://github.com/Azure/azure-cli/issues/21815

@kangsun-ctrl Could you show us the specific code link? I don't find any code related to `eula` in https://github.com/Azure/azure-cli/pull/3

**Comment by zhoxing-ms** [comment]
@kangsun-ctrl By the way, since the community gallery image related commands are still in `image-gallery` extension, it is not convenient to create scenario tests in the main repo. 
Could you help test the private package generated by the edge build before we release this feature to see whether the effect meets your expectation?

File: src/azure-cli/azure/cli/command_modules/vm/_help.py [hunk 1]
```diff
@@ -2841,6 +2844,9 @@
   - name: Create a VMSS from shared gallery image. (private preview feature, please contact shared image gallery team by email sigpmdev@microsoft.com to register for preview if you're interested in using this feature).
     text: >
         az vmss create -n MyVmss -g MyResourceGroup --image /SharedGalleries/{gallery_unique_name}/Images/{image}/Versions/{version}
+  - name: Create a VMSS from community gallery image. (private preview feature, please contact community image gallery team by email sigpmdev@microsoft.com to register for preview if you're interested in using this feature).
```
Comment (wangzelin007) [inline src/azure-cli/azure/cli/command_modules/vm/_help.py:2847]:
```suggestion
  - name: Create a VMSS from community gallery image (private preview feature, please contact community image gallery team by email sigpmdev@microsoft.com to register for preview if you're interested in using this feature).
```

File: src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py [hunk 2]
```diff
@@ -397,6 +397,34 @@ def parse_shared_gallery_image_id(image_reference):
...
+
+def parse_community_gallery_image_id(image_reference):
+    from azure.cli.core.azclierror import InvalidArgumentValueError
+
+    if not image_reference:
+        raise InvalidArgumentValueError(
+            'Please pass in the community gallery image id through the parameter --image')
+
+    image_info = re.search(r'^/CommunityGalleries/([^/]*)/Images/([^/]*)/Versions/.*$', image_reference, re.IGNORECASE)
+    if not image_info or len(image_info.groups()) < 2:
```
Comment (wangzelin007) [inline src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py:419]: Why not check image version ? 
```suggestion
    image_info = re.search(r'^/CommunityGalleries/([^/]*)/Images/([^/]*)/Versions/(.*)$', image_reference, re.IGNORECASE)
    if not image_info or len(image_info.groups()) < 3:
```

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py:419]: At present, we need to parse the `public_gallery_name` and `gallery_image_name` from community gallery image id to query the community gallery image info, but version is not used for the time being, so we do not further match version.
https://github.com/Azure/azure-cli/blob/ddb1665b7baa01fe3ec7d75b846a4c89a2d832d6/src/azure-cli/azure/cli/command_modules/vm/_validators.py#L565-L570

I think we can add version matching when we need to parse the version. What do you think?

**Review by wangzelin007 — APPROVED** [review]

**Pull request merged**